sympy/tensor/functions.py

```diff
@@ -89,14 +89,14 @@ def shape(expr):
 
     This function returns the shape of any object representing matrix or array.
 
-    >>> from sympy import shape, Array, Matrix, Integral
+    >>> from sympy import shape, Array, ImmutableDenseMatrix, Integral
     >>> from sympy.abc import x
     >>> A = Array([1, 2])
     >>> shape(A)
     (2,)
     >>> shape(Integral(A, x))
     (2,)
-    >>> M = Matrix([1, 2])
+    >>> M = ImmutableDenseMatrix([1, 2])
```

Comment:
Shouldn't this get sympified somewhere?

Reply:
It does in `Integral`, but not in `NewExpr` (user defined subclass). Here I thought the doc would be easier to read if `NewExpr` didn't have a `__new__` method defined

Reply:
Ok, fair enough